### PR TITLE
Fix energy consumption limit bug

### PR DESCRIPTION
**Pull request by gustavo-1**
OP#74

### CODE REVIEW

File: src/ispd/gui/MainWindow.java [hunk 1]
```diff
@@ -748,7 +748,7 @@ private void initFileChooser() {
         this.jFileChooser.setAcceptAllFileFilterUsed(false);
         this.jFileChooser.setFileFilter(this.fileFilter);
         this.jFileChooser.setFileView(new IspdFileView());
-        this.jFileChooser.setSelectedFile(this.configure.getLastFile());
+        //this.jFileChooser.setSelectedFile(this.configure.getLastFile());
```
Comment (MathRPG) [inline src/ispd/gui/MainWindow.java:751]: Do not leave commented-out code in the repository.

File: src/ispd/gui/UserConfigurationDialog.java [hunk 2]
```diff
@@ -149,18 +149,22 @@ private void onAddClick(final ActionEvent evt) {
         );
 
         // TODO: Parse result and error.
-        final Double result =
-                Double.parseDouble(JOptionPane.showInputDialog(this,
-                "Enter user power comsumption limit"));
+        String sResult ="";
```
Comment (MathRPG) [inline src/ispd/gui/UserConfigurationDialog.java]: Merge declaration and initialization in one line; make variable 'final' if applicable.

File: src/ispd/gui/UserConfigurationDialog.java [hunk 3]
```diff
@@ -149,18 +149,22 @@ private void onAddClick(final ActionEvent evt) {
...
+        sResult =JOptionPane.showInputDialog(this,
+                        "Enter user power comsumption limit");
 
-        if (!this.userSet.contains(add) && !add.isEmpty()) {
+
+        if (!this.userSet.contains(add) && !add.isEmpty() && !sResult.isEmpty() && !this.userSet.contains(sResult) ){
```
Comment (MathRPG) [inline src/ispd/gui/UserConfigurationDialog.java]: Overly-complex expression, extract this section of logic away into a method.

**Review by MathRPG — CHANGES_REQUESTED** [review]
Remember to write commit descriptions in English, even if they're planned to be squashed. It helps make the repository consistent.

File: src/ispd/gui/UserConfigurationDialog.java [hunk 4]
```diff
@@ -141,22 +141,21 @@ private String translate(final String word) {
...
         // TODO: Parse result and error.
-        String sResult =JOptionPane.showInputDialog(this,
+        final String sResult =JOptionPane.showInputDialog(this,
                         "Enter user power comsumption limit");
 
 
-        if (add != null && sResult != null && !add.isEmpty() && !this.userSet.contains(add)  && !sResult.isEmpty() && !this.userSet.contains(sResult) ){
-            final Double result =
-                    Double.parseDouble(sResult);
+        if (!add.isEmpty() && !this.userSet.contains(add)  && !sResult.isEmpty() && !this.userSet.contains(sResult) ){
```
Comment (MathRPG) [inline src/ispd/gui/UserConfigurationDialog.java:157]: Have you tested what happens when you cancel the input (i.e., close the prompt window)? Make sure it is working as intended.

File: src/ispd/gui/UserConfigurationDialog.java [hunk 5]
```diff
@@ -149,18 +150,23 @@ private void onAddClick(final ActionEvent evt) {
...
+            final Double result = Double.parseDouble(sResult);
             this.userSet.add(add);
             final Vector user = new Vector<String>();
             user.add(add);
             user.add(result);
             this.users.add(user);
             this.jScrollPane1.setViewportView(this.table);
+        }else{
+            JOptionPane.showMessageDialog(null, "Insert values corectly");
```
Comment (barrosoman) [inline src/ispd/gui/UserConfigurationDialog.java:166]: A typo in `corectly`, should be `correctly`.

File: src/ispd/gui/UserConfigurationDialog.java [hunk 6]
```diff
@@ -149,18 +150,23 @@ private void onAddClick(final ActionEvent evt) {
...
-        final Double result =
-                Double.parseDouble(JOptionPane.showInputDialog(this,
-                "Enter user power comsumption limit"));
+        final String sResult =JOptionPane.showInputDialog(this,
+                        "Enter user power comsumption limit");
+
 
-        if (!this.userSet.contains(add) && !add.isEmpty()) {
+        if (!add.isEmpty() && !this.userSet.contains(add)  && !sResult.isEmpty() && !this.userSet.contains(sResult) ){
+            final Double result = Double.parseDouble(sResult);
```
Comment (barrosoman) [inline src/ispd/gui/UserConfigurationDialog.java:158]: Test when the `sResult` is not a valid double, should be enclosed in a try/catch block.

**Review by MathRPG — CHANGES_REQUESTED** [review]
These review notes were talked about in person; but I'm documenting them here on github for consistency.

**Comment by MathRPG** [comment]
Closed due to the GitLab migration

**Pull request closed without merging**